lib/hamster/hash.rb

```diff
@@ -590,6 +590,25 @@ def fetch_values(*wanted)
       Vector.new(array.freeze)
     end
 
+    # Return the value of successively indexing into a hash
+    # If any of the keys is not present in the hash, return `nil`
+    #
+    # @example
+    #   h = Hamster::Hash[:a => 9, :b => Hamster::Hash[:c => 'a', :d => 4], :e => nil]
+    #   h.dig(:b, :c) # => "a"
+    #   h.dig(:b, :f) # => nil
+    # @params keys to fetch from the hash
+    # @return [Object]
+    def dig(key, *rest)
+      value = self[key]
+      if rest.empty? || value.nil?
+        value
+      elsif value.respond_to?(:dig)
+        value.dig(*rest)
```

Comment:
Out of curiosity: is this how MRI's `#dig` also works? If it finds a value which does not respond to `dig`, does it error out, or just return `nil`?